third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -220,6 +225,15 @@ LogicalResult StreamPipeliner::initSchedule(int maxIndirectionLevel) {
   bool pairedGlobalLoadLocalStore = stages[SCHED_LOCAL_STORE] == 0;
   stages[SCHED_LOCAL_STORE] += maxIndirectionLevel;
 
+  // In useAsyncCopy + PingPong case, we'd want to hoist out first async_wait
+  // out of the loop, and async_wait within the loop be towards the end.
+  // This is beneficial for maximizing hiding of latency, while ensuring
+  // 2 barriers between asyncWait and localLoad at start of loop S.T
+  // we do not hit race conditions between warp-lo and warp-hi.
+  if (useAsyncCopy && useBlockPingpong) {
+    stages[SCHED_ASYNC_WAIT] = stages[SCHED_LOCAL_STORE];
```

Comment:
I think this should be `.. = max(0, stages[SCHED_LOCAL_LOAD] - 1)` to work correctly with num_stages  > 2. We always want the AsyncWait one stage before the LocalLoads.

Reply:
ah nice! will do that :)

Reply:
done!